src/Group.php

```diff
@@ -1083,4 +1103,56 @@ public function getGroupLink(bool $enable_anonymization = false): string
 
         return $this->getLink();
     }
+
+    public function post_addItem()
+    {
+        // Adding a new group might invalidate the group cache if it's a new child
+        // group and recursive membership is enabled
+        if ($this->fields['groups_id']) {
+            Group::updateLastGroupChange();
+        }
+    }
+
+    public function post_updateItem($history = 1)
+    {
+        // Changing a group's parent might invalidate the group cache if recursive
+        // membership is enabled
+        if (
+            isset($this->oldvalues['groups_id'])
+            && $this->fields['groups_id'] !== $this->oldvalues['groups_id']
+        ) {
+            Group::updateLastGroupChange();
+        }
+
+        // Enabling or disabling recursion on a group will invalidate the group
+        // cache
+        if (
+            isset($this->oldvalues['recursive_membership'])
+            && $this->fields['recursive_membership'] !== $this->oldvalues['recursive_membership']
+        ) {
+            Group::updateLastGroupChange();
+        }
```

Comment:
Could be groupped into a single `if` block, to prevent updating twice the cached value when both properties are changed at the same time.